TACC-Cloud/geoapi
# DES-2182: Questionnaire asset type

## Overview: ##
Add RAPP questionnaire file (`.rq`) as an importable asset type.

## PR Status: ##


## Related Jira tickets: ##

* [DES-2182](https://jira.tacc.utexas.edu/browse/DES-2182)
* [DES-2380](https://jira.tacc.utexas.edu/browse/DES-2380) (partially)

## Summary of Changes: ##
Added questionnaire asset type (panel icon, allowed assets).
Added questionnaire viewer.

## Testing Steps: ##
1. Run related geoapi https://github.com/TACC-Cloud/hazmapper/pull/118 
2. 1. Import `.rq` file from [test project](https://www.designsafe-ci.org/data/browser/projects/8649738474506818026-242ac116-0001-012/) 
3. Ensure that questionnaire assets display properly in asset panel
4. Ensure that questionnaire viewer renders properly.

## Notes: ##
Concerns:
1. The questionnaire viewer code provided by RAPID team was tightly coupled with the render code, which uses jquery. So I had jquery installed for now. But am still tryiing to look for options to isolate jquery.
3. Used `::ng-host` to properly display the `questionnaireBuilder.css` code (globally accessible css)
